File: avalon-modules/avalon-dataset/avalon-dataset-service/src/main/java/io/github/lishangbu/avalon/dataset/service/BerryFirmnessService.java
package io.github.lishangbu.avalon.dataset.service;

import io.github.lishangbu.avalon.dataset.entity.BerryFirmness;
import java.util.List;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

/// 树果坚硬度服务。
public interface BerryFirmnessService {

    /// 根据条件分页查询树果坚硬度。
    Page<BerryFirmness> getPageByCondition(BerryFirmness berryFirmness, Pageable pageable);

    /// 新增树果坚硬度。
    BerryFirmness save(BerryFirmness berryFirmness);

    /// 更新树果坚硬度。
    BerryFirmness update(BerryFirmness berryFirmness);

    /// 根据主键删除树果坚硬度。
    void removeById(Long id);

    /// 根据条件查询树果坚硬度列表。
    List<BerryFirmness> listByCondition(BerryFirmness berryFirmness);
}

File: avalon-modules/avalon-dataset/avalon-dataset-service/src/main/kotlin/io/github/lishangbu/avalon/dataset/service/BerryFirmnessService.kt
package io.github.lishangbu.avalon.dataset.service

import io.github.lishangbu.avalon.dataset.entity.dto.BerryFirmnessSpecification
import io.github.lishangbu.avalon.dataset.entity.dto.BerryFirmnessView
import io.github.lishangbu.avalon.dataset.entity.dto.SaveBerryFirmnessInput
import io.github.lishangbu.avalon.dataset.entity.dto.UpdateBerryFirmnessInput
import org.babyfish.jimmer.Page
import org.springframework.data.domain.Pageable

/** 树果硬度服务 */
interface BerryFirmnessService {
    /** 按条件分页查询树果硬度*/
    fun getPageByCondition(
        specification: BerryFirmnessSpecification,
        pageable: Pageable,
    ): Page<BerryFirmnessView>

    /** 保存树果硬度 */
    fun save(command: SaveBerryFirmnessInput): BerryFirmnessView

    /** 更新树果硬度 */
    fun update(command: UpdateBerryFirmnessInput): BerryFirmnessView

    /** 按 ID 删除树果硬度 */
    fun removeById(id: Long)

    /** 根据条件查询树果硬度列表 */
    fun listByCondition(specification: BerryFirmnessSpecification): List<BerryFirmnessView>
}
